modules/integration/src/test/scala/scala/cli/integration/PackageTestDefinitions.scala

```diff
@@ -469,10 +469,60 @@ abstract class PackageTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
-  if (!Properties.isWin && actualScalaVersion.startsWith("2.13"))
+  def libraryNativeTest(shared: Boolean = false): Unit = {
+    val fileName     = "simple.sc"
+    val nativeTarget = if (shared) "dynamic" else "static"
+    val inputs = TestInputs(
+      os.rel / fileName ->
+        s"""
+           |//> using platform scala-native
+           |//> using nativeTarget $nativeTarget
+           |import scala.scalanative.unsafe._
+           |object myLib{
+           |  @exported
+           |  def addLongs(l: Long, r: Long): Long = l + r
+           |  @exported("mylib_addInts")
+           |  def addInts(l: Int, r: Int): Int = l + r
+           |}""".stripMargin
+    )
+    val destName = {
+      val ext =
+        if (!shared)
+          if (Properties.isWin) ".lib" else ".a"
+        else if (Properties.isWin) ".dll"
+        else if (Properties.isMac) ".dylib"
+        else ".so"
+      fileName.stripSuffix(".sc") + ext
+    }
+
+    inputs.fromRoot { root =>
+      os.proc(TestUtil.cli, "--power", "package", extraOptions, fileName).call(
+        cwd = root,
+        stdin = os.Inherit,
+        stdout = os.Inherit
+      )
+
+      val library = root / destName
+      expect(os.isFile(library))
+    }
+  }
+
+  if (!Properties.isWin && actualScalaVersion.startsWith("2.13")) {
     test("simple native") {
       simpleNativeTest()
     }
+    test("dynamic library native") {
+      libraryNativeTest(shared = true)
+    }
+
+    // To produce a static library, `LLVM_BIN` environment variable needs to be
+    // present (for `llvm-ar` utility)
+    if (sys.env.contains("LLVM_BIN"))
```

Comment:
hey, for this to run on the CI we'd have to have the utility available and have the env variable point to it, right?
Am I assuming correctly that this test is being skipped on the CI right now?

Reply:
Yes, I explicitly ignored this test for now to save you the pain of installing LLVM on CI :D 

Additionally, I'm working on not needing this at all: https://github.com/scala-native/scala-native/pull/3548

Once that's merged and hopefully backported, we can remove the LLVM_BIN check and things should work out of the box

Reply:
ah, okay.
I was already looking into setting it up on CI ;D
cool, in that case it's okay to skip this test temporarily 👍

Reply:
If you'd prefer - I can set up LLVM in only one of the workflows that runs this test (just need to figure out which one), to make sure we don't ignore any CI tests unnecessarily

Reply:
If you have the time to set up the workflow, go ahead.
Otherwise, it's also fine to leave it temporarily inactive, since it seems it won't be necessary anyway once https://github.com/scala-native/scala-native/pull/3548 gets merged.

Reply:
> Maybe easier that way for now

Ok, I won't do it then :)

Once we have a timeline/decision on that PR being backported in 0.4.x, I can go ahead and make a separate PR adding LLVM (if necessary).